File: php/create_account.php
<?php

	//store user information in database
	//check if email address already exists before trying to store information
	
	$json = file_get_contents('php://input');
	$object = json_decode($json);
	
	$first_name = $object->first;
	$last_name = $object->last;
	$user = $object->user;
	$pw = $object->password;
	
	$response["successIndicator"] = "neutral";
	
	include 'db_connect.php';
	
	//check if provided email address already exists in the database 
	$duplicate_query = "SELECT * FROM useraccount WHERE email_address = '$user'";
	$result_duplicate = mysqli_query($link, $duplicate_query);
	
	if(mysqli_num_rows($result_duplicate) > 0){
		
		$response["successIndicator"] = "exists";
		
	}else{
		
		$query = "INSERT INTO useraccount VALUES('$user', '$first_name', '$last_name', '$pw', now())";
		$result = mysqli_query($link, $query);
		
		if(!empty($result)){
			
			$response["successIndicator"] = "success";
			
		}else{
			
			$response["successIndicator"] = "failure";
		}
			
	}
		
	echo json_encode($response);
		
	
	
	
	
	
	
		
	
	

	/*
	$response = array();
	
	if(isset($_POST['firstname']) && isset($_POST['lastname']) && isset($_POST['emailaddress']) && isset($_POST['pass'])){
		
		$first = $_POST['firstname'];
		$last = $_POST['lastname'];
		$email = $_POST['emailaddress'];
		$password = $_POST['pass'];
		
		require_once __DIR__.'db_connect.php';
		
		$db = new DB_CONNECT();
		
		//check if provided email address already exists in the database 
		$duplicate_query = "SELECT email_address FROM useraccount WHERE email_address = '$email'";
		$result_duplicate = mysqli_query($duplicate_query);
		
		//only store if email address is unique
		if(empty($result_duplicate)){
			
			$query = "INSERT INTO useraccount VALUES($email, $first, $last, $password, now())";
			$result = mysqli_query($query);
			
			//return success if information has been stored successfully
			if(!empty($result)){
				
				$response['success'] = 1;
				$response['message'] = "User information was successfully stored."
				
				echo json_encode($response);
				
			}else{
				
				$response['success'] = 0;
				$response['message'] = "Error. User information was not stored."
				
				echo json_encode($response);
				
			}
			
		}else{
			
			//email address provided already exists, so do not store information
			$response['success'] = 0;
			$response['message'] = "User email address already exists. User information was not stored."
				
			echo json_encode($response);
		}		
		
	}
	*/

?>

File: servlets/CreateAccount.java
import java.io.BufferedReader;
import java.io.IOException;
import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.util.logging.*;
import org.json.JSONException;
import org.json.JSONObject;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;
import java.sql.Timestamp;

/**
 * Servlet implementation class CreateAccount
 */

@WebServlet("/CreateAccount")
public class CreateAccount extends HttpServlet {
	
	private static final long serialVersionUID = 1L;
	private static final Logger LOGGER = Logger.getLogger("InfoLogging");
       
    public CreateAccount() {}

	protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {}

	protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
		
		String line = null;
		String firstName, lastName, userEmail, userPassword;
		String result = "";
		String responseToApp = "neutral";
		Connection connection = null;
		Statement duplicateQueryStatement = null;
		PreparedStatement queryStatement = null;
		ResultSet resultSet = null;
		
		try {
			
			BufferedReader reader = request.getReader();
			
			while((line = reader.readLine()) != null) {
				result += line;
			}
			
			JSONObject jsonObject = new JSONObject(result);
			
			// retrieve sent user data
			firstName = jsonObject.getString("first");
			lastName = jsonObject.getString("last");
			userEmail = jsonObject.getString("user");
			userPassword = jsonObject.getString("password");
			
			if(!firstName.equals("")) {
				firstName = firstName.substring(0,1).toUpperCase() + firstName.substring(1);
			}
			
			if(!lastName.equals("")) {
				lastName = lastName.substring(0, 1).toUpperCase() + lastName.substring(1);
			}
			
			// Establish connection with database
			Class.forName("com.mysql.jdbc.Driver");
			connection = DriverManager.getConnection("jdbc:mysql://173.244.1.42:3306/S0280202", "S0280202", "New2018");
			
			// check if email address already exists
			String sqlString = "SELECT * FROM useraccount WHERE email_address = '" + userEmail + "'";
			duplicateQueryStatement = connection.createStatement();
			resultSet = duplicateQueryStatement.executeQuery(sqlString);
			
			// if this user does not have an account yet, insert information into database
			if(!resultSet.next()) {
				
				Timestamp timestamp = new Timestamp(System.currentTimeMillis());
				
				String sqlQueryString = "INSERT INTO useraccount VALUES(?, ?, ?, ?, ?)";
				queryStatement = connection.prepareStatement(sqlQueryString);
				
				queryStatement.setString(1, userEmail);
				queryStatement.setString(2,  firstName);
				queryStatement.setString(3, lastName);
				queryStatement.setString(4, userPassword);
				queryStatement.setTimestamp(5, timestamp);
				
				queryStatement.executeUpdate();
				
				responseToApp = "success";
				
			}else {
				responseToApp = "exists";
			}
			
			// send response back to client
			JSONObject responseObject = new JSONObject();
			responseObject.put("successIndicator", responseToApp);
			String json = responseObject.toString();
			
			response.setContentType("application/json");
			response.setCharacterEncoding("UTF-8");
			response.getWriter().write(json);
			
		}catch(SQLException e) {
			LOGGER.info("Error sqlexception");
			e.printStackTrace();	
		}catch(JSONException ex) {
			LOGGER.info("Error jsonexception");
			ex.printStackTrace();
		}catch(ClassNotFoundException en) {
			LOGGER.info("Error class not found exception");
			en.printStackTrace();
		}finally {
			try {		
				if(duplicateQueryStatement != null) {
					duplicateQueryStatement.close();
				}
				if(queryStatement != null) {
					queryStatement.close();
				}
				if(resultSet != null) {
					resultSet.close();
				}
				if(connection != null) {
					connection.close();
				}
			}catch(SQLException sqlEx) {
				sqlEx.printStackTrace();
			}
		}
	}
}

